docs/topics/consumers.rst

```diff
@@ -277,3 +281,66 @@ AsyncJsonWebsocketConsumer
 An async version of ``JsonWebsocketConsumer``, available as
 ``channels.generic.websocket.AsyncJsonWebsocketConsumer``. Note that even
 ``encode_json`` and ``decode_json`` are async functions.
+
+
+AsyncHttpConsumer
+~~~~~~~~~~~~~~~~~
+
+Available as ``channels.generic.http.AsyncHttpConsumer``, this offers basic
+primitives to implement a HTTP endpoint::
+
+    from channels.generic.http import AsyncHttpConsumer
+
+    class BasicHttpConsumer(AsyncHttpConsumer):
+        async def handle(self, body):
+            await asyncio.sleep(10)
+            await self.send_response(200, b"Your response bytes", headers=[
+                ("Content-Type", "text/plain"),
+            ])
+
+You are expected to implement your own ``self.handle`` method. The
+method receives the whole request body as a single bytestring.  Headers
+may either be passed as a list of tuples or as a dictionary. The
+response body content is expected to be a bytestring.
+
+If you need more control over the response, e.g. for implementing long
+polling, use the lower level ``self.send_headers`` and ``self.send_body``
+methods instead. This example already mentions channel layers which will
+be explained in detail later::
+
+    import json
+    from channels.generic.http import AsyncHttpConsumer
+
+    class LongPollConsumer(AsyncHttpConsumer):
+        async def handle(self, body):
+            await self.send_headers(headers=[
+                ("Content-Type", "application/json"),
+            ])
+            # Headers are only sent after the first body event.
+            # Set "more_body" to tell the interface server to not
+            # finish the response yet:
+            await self.send_body(b"", more_body=True)
+
+        async def chat_message(self, event):
+            # Send JSON and finish the response:
+            await self.send_body(json.dumps(event).encode("utf-8"))
+
+Of course you can also use those primitives to implement a HTTP endpoint for
+`Server-sent events`_::
```

Comment:
I'd prefer this as an inline link (I know, I'm weird, but the rest of the docs work like that and I like consistency)